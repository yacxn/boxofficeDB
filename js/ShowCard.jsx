import React from "react";
import { shape, string } from "prop-types";
import styled from "styled-components";

const Wrapper = styled.div`
width: 32%;
border: 2px solid #333;
border-radius: 4px;
margin-bottom: 25px;
padding-right: 10px;
overflow:hidden;

`;
const Image = styled.img`
width: 46%;
float: left;
margin-right: 10px;
`;

const ShowCard = props => (
  <Wrapper className="props.show-card">
    <Image
      src={`/public/img/posters/${props.poster}`}
      alt={`${props.title} Show Poster`}
    />
    <div>
      <h3>{props.title}</h3>
      <h4>{props.title}</h4>
    </div>
  </Wrapper>
);

ShowCard.PropTypes = {
  poster: string.isRequired,
  title: string.isRequired,
  year: string.isRequired,
  description: string.isRequired
};
export default ShowCard;
