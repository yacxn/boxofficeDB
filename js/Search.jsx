import React, { Component } from "react";
import ShowCard from "./ShowCard";
import preload from "../data.json";

class Search extends Component {
  state = {
    searchTerm: ''
  };
  handleSearchTermChange = event => {
    this.setState({ searchTerm: event.target.value });
  }
  render() {
    return (
      <div className="search">
        <header>
          <h1>Myvid.</h1>
          <input
            type="text"
            onChange={this.handleSearchTermChange}
            value={this.state.searchTerm}
            placeholder="Search"
          />
        </header>
        <div>
          {preload.shows
            .filter(show => `${show.title} ${show.description}`.toUpperCase().indexOf(this.state.searchTerm.toUpperCase()) >= 0
          )
            .map(show => <ShowCard key={show.imdbID} {...show} />)}
        </div>
      </div>
    );
  }
}
export default Search;
