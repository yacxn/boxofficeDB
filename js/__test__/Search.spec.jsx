import React from "react";
import { shallow } from 'enzyme';
import preload from '../../data.json';
import Search from "../Search";
import ShowCard from '../ShowCard';

test("Search renders correctly", () => {
  const component = shallow(<Search />);
  expect(component).toMatchSnapshot();
});
test('Search should render correct amout of shows', ()
  => {
    const component = shallow(<Search />)
  expect(component.find(ShowCard).length).toEqual(preload.shows.length);
})
test('Search should render correct amout of shows based on search term', ()
=> {

});
