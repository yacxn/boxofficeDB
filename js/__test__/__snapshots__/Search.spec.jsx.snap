// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="search"
>
  <header>
    <h1>
      Myvid.
    </h1>
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </header>
  <div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Atlanta Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/a.jpg"
      />
      <div>
        <h3>
          Atlanta
        </h3>
        <h4>
          Atlanta
        </h4>
        <p>
          Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; "Earnest 'Earn' Marks," an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Billions Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/b.jpg"
      />
      <div>
        <h3>
          Billions
        </h3>
        <h4>
          Billions
        </h4>
        <p>
          U.S. Attorney Chuck Rhoades goes after hedge fund king, Bobby "Axe" Axelrod in a battle between two powerful New York figures.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Black Mirror Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/bm.jpg"
      />
      <div>
        <h3>
          Black Mirror
        </h3>
        <h4>
          Black Mirror
        </h4>
        <p>
          A television anthology series that shows the dark side of life and technology.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Breaking Bad Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/bb.jpg"
      />
      <div>
        <h3>
          Breaking Bad
        </h3>
        <h4>
          Breaking Bad
        </h4>
        <p>
          A high school chemistry teacher diagnosed with inoperable lung cancer turns to manufacturing and selling methamphetamine in order to secure his family's future.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Game of Thrones Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/got.jpg"
      />
      <div>
        <h3>
          Game of Thrones
        </h3>
        <h4>
          Game of Thrones
        </h4>
        <p>
          Nine noble families fight for control over the mythical lands of Westeros, while a forgotten race returns after being dormant for thousands of years.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Homeland Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/h.jpg"
      />
      <div>
        <h3>
          Homeland
        </h3>
        <h4>
          Homeland
        </h4>
        <p>
          A bipolar CIA operative becomes convinced a prisoner of war has been turned by al-Qaeda and is planning to carry out a terrorist attack on American soil.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="House of Cards Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/hoc.jpg"
      />
      <div>
        <h3>
          House of Cards
        </h3>
        <h4>
          House of Cards
        </h4>
        <p>
          A Congressman works with his equally conniving wife to exact revenge on the people who betrayed him.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Master of None Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/mon.jpg"
      />
      <div>
        <h3>
          Master of None
        </h3>
        <h4>
          Master of None
        </h4>
        <p>
          The personal and professional life of Dev, a 30-year-old actor in New York.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Narcos Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/n.jpg"
      />
      <div>
        <h3>
          Narcos
        </h3>
        <h4>
          Narcos
        </h4>
        <p>
          A chronicled look at the criminal exploits of Colombian drug lord Pablo Escobar.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Orange Is the New Black Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/oitnb.jpg"
      />
      <div>
        <h3>
          Orange Is the New Black
        </h3>
        <h4>
          Orange Is the New Black
        </h4>
        <p>
          The story of Piper Chapman, a woman in her thirties who is sentenced to fifteen months in prison after being convicted of a decade-old crime of transporting money for her drug-dealing girlfriend.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Silicon Valley Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/sv.jpg"
      />
      <div>
        <h3>
          Silicon Valley
        </h3>
        <h4>
          Silicon Valley
        </h4>
        <p>
          Follows the struggle of Richard Hendricks, a silicon valley engineer trying to build his own company called Pied Piper.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Stranger Things Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/st.jpg"
      />
      <div>
        <h3>
          Stranger Things
        </h3>
        <h4>
          Stranger Things
        </h4>
        <p>
          When a young boy disappears, his mother, a police chief, and his friends must confront terrifying forces in order to get him back.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="The Americans Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/ta.jpg"
      />
      <div>
        <h3>
          The Americans
        </h3>
        <h4>
          The Americans
        </h4>
        <p>
          Two Soviet intelligence agents pose as a married couple to spy on the American government.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Veep Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/v.jpg"
      />
      <div>
        <h3>
          Veep
        </h3>
        <h4>
          Veep
        </h4>
        <p>
          Former Senator Selina Meyer finds that being Vice President of the United States is nothing like she hoped and everything that everyone ever warned her about.
        </p>
      </div>
    </div>
    <div
      className="props.show-card sc-bdVaJa iBHuow"
    >
      <img
        alt="Westworld Show Poster"
        className="sc-bwzfXH bbXcwM"
        src="/public/img/posters/ww.jpg"
      />
      <div>
        <h3>
          Westworld
        </h3>
        <h4>
          Westworld
        </h4>
        <p>
          Set at the intersection of the near future and the reimagined past, explore a world in which every human appetite, no matter how noble or depraved, can be indulged without consequence.
        </p>
      </div>
    </div>
  </div>
</div>
`;
